connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -703,6 +703,19 @@ public Config describeTopicConfig(String topic) {
         return result;
     }
 
+    protected Map<TopicPartition, Long> retryEndOffsets(Set<TopicPartition> partitions, Duration timeoutDuration, long retryBackoffMs) {
+
+        try {
+            return RetryUtil.retryUntilTimeout(
+                    () -> endOffsets(partitions),
+                    () -> "list offsets",
+                    timeoutDuration,
+                    retryBackoffMs);
+        } catch (Exception e) {
+            throw new ConnectException("Failed to read offsets for topic partitions.", e);
+        }
```

Comment:
We should use consisting phrasing, and actually "list offsets for topic partitions" most closely aligns with the admin client call that `endOffsets(...)` is using. So how about:
```suggestion
        try {
            return RetryUtil.retryUntilTimeout(
                    () -> endOffsets(partitions),
                    () -> "list offsets for topic partitions",
                    timeoutDuration,
                    retryBackoffMs);
        } catch (Exception e) {
            throw new ConnectException("Failed to list offsets for topic partitions.", e);
        }
```
Also, using `list offsets for topic partitions` as the retry function description works quite well with the log and exception message within `retryUntilTimeout(...)` method.